src/components/atoms/VideoOverlayButton/VideoOverlayButton.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect } from "react";
+import { useToggle } from "react-use";
+import {
+  faMicrophone,
+  faMicrophoneSlash,
+  faVideo,
+  faVideoSlash,
+  faVolumeMute,
+  faVolumeUp,
+  IconDefinition,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import classNames from "classnames";
+
+import { ContainerClassName } from "types/utility";
+
+import { assertUnreachable } from "utils/error";
+
+import "./VideoOverlayButton.scss";
+
+type VideoOverlayButtonVariant = "microphone" | "camera" | "audio";
+
+export interface VideoOverlayButtonProps extends ContainerClassName {
+  variant: VideoOverlayButtonVariant;
+  defaultValue: boolean;
+  onEnabledChanged: (enabled: boolean) => void;
+}
+
+export const VideoOverlayButton: React.FC<VideoOverlayButtonProps> = ({
+  variant,
+  defaultValue,
+  onEnabledChanged,
+  containerClassName,
+}) => {
+  const [enabled, toggle] = useToggle(defaultValue);
+
+  useEffect(() => {
+    onEnabledChanged(enabled);
+  }, [enabled, onEnabledChanged]);
```

Comment:
```suggestion
  useEffect(() => onEnabledChanged(enabled), [enabled, onEnabledChanged]);
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2221/commits/06234506347c70293948d6367762654d711fcc4b